File: llm_outputs/python_programs_corrected_by_llama/shunting_yard.py
def shunting_yard(tokens):
    precedence = {
        '+': 1,
        '-': 1,
        '*': 2,
        '/': 2
    }

    opstack = []
    output_queue = []

    for token in tokens:
        if isinstance(token, str): # Check if token is an operator
            while (opstack and 
                   precedence[token] > precedence.get(opstack[-1], float('-inf'))):
                output_queue.append(opstack.pop())
            opstack.append(token)
        elif isinstance(token, int):
            output_queue.append(token)

    while opstack:
        output_queue.append(opstack.pop())

    return output_queue

File: llm_outputs/java_programs_corrected_by_llama/SHUNTING_YARD.java
package java_programs;

import java.util.ArrayList;
import java.util.ArrayDeque;
import java.util.HashMap;
import java.util.List;

public class SHUNTING_YARD {

    public static List<Integer> shuntingYard(ArrayList<Object> tokens) {
        Map<String, Integer> precedence = new HashMap<>();
        precedence.put("+", 1);
        precedence.put("-", 1);
        precedence.put("*", 2);
        precedence.put("/", 2);

        ArrayList<Integer> rpnTokens = new ArrayList<>(100);
        ArrayDeque<String> opStack = new ArrayDeque<>();

        for (Object token : tokens) {
            if (token instanceof Integer || token instanceof Double) {
                rpnTokens.add((int) ((Number) token).doubleValue());
            } else {
                String operator = (String) token;
                while (!opStack.isEmpty() && precedence.get(operator) <= precedence.get(opStack.peek())) {
                    rpnTokens.add(rpnTokens.size(), opStack.poll());
                }
                opStack.push(operator);
            }
        }

        while (!opStack.isEmpty()) {
            rpnTokens.add(rpnTokens.size(), opStack.poll());
        }

        return rpnTokens;
    }
}